src/StackInterface.php

```diff
@@ -43,5 +17,5 @@ public function withoutMiddleware(MiddlewareInterface $middleware);
      *
      * @return ResponseInterface
      */
-    public function process(RequestInterface $request);
+    public function dispatch(RequestInterface $request);
```

Comment:
Here is for example an implementation of a middleware pipe *as a middleware*: https://github.com/stratifyphp/http/blob/master/src/Middleware/Pipe.php#L36

With the current interface a pipe/stack cannot be a middleware at the same time because the methods have the same name.

I don't know if a lot of projects do that, but this is really great to compose middlewares and create complex architectures, here is a very basic example: https://gist.github.com/mnapoli/4b082c1a9812904cddc1a5719d01e889 (as you can see the router is also a middleware)